lib/src/local_working_copy.rs

```diff
@@ -1034,15 +1046,22 @@ impl TreeState {
                         && git_ignore.matches(path.as_internal_file_string())
                     {
                         // If it wasn't already tracked and it matches
-                        // the ignored paths, then
-                        // ignore it.
+                        // the ignored paths, then ignore it.
                     } else {
+                        if maybe_current_file_state.is_none()
+                            && !start_tracking_matcher.matches(&path)
+                        {
```

Comment:
nit: I would move this one level up as `} else if ..`. This is quite similar to `git_ignore.matches()`.